### PR TITLE
Updating pipeline linux agent image to Ubuntu 18.04

### LINKED ISSUE #17773: Update docker image of linux build agents
The current linux build agents use Ubuntu 16 which is out of LTS window on April 30th 2021. We need to update our build agents to use the latest version of Ubuntu.

### PR DESCRIPTION
**Pull request by aasimkhan30**
Doing this update since Ubuntu 16.04 used in the current linux builds has passed EOL 

This PR fixes #17773

Link to the adhoc build: 
https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=131021&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=34fc863e-00ed-50b5-30ee-f50930ed6e81

### CODE REVIEW

File: build/azure-pipelines/linux/Dockerfile
Comment (Charles-Gagnon) [inline build/azure-pipelines/linux/Dockerfile:4]: So the process is update Docker file -> push update to the container registry -> update pipeline?

**Review by Charles-Gagnon — APPROVED** [review]
It looks fine but I'd suggest we wait until the release branch is cutoff before merging this in just to avoid potential issues.

**Review by alanrenmsft — APPROVED** [review]

File: build/azure-pipelines/linux/Dockerfile
Comment (aasimkhan30) [reply build/azure-pipelines/linux/Dockerfile:4]: Yes, 

1. Update the docker file
1. Build an image 
1. Push to container registry 
1. Update the pipeline. 

I will create a doc for this as well

**Pull request merged**
